public/app/plugins/datasource/cloud-monitoring/functions.ts

```diff
@@ -125,3 +133,9 @@ export const alignmentPeriodLabel = (customMetaData: CustomMetaData, datasource:
   const hms = rangeUtil.secondsToHms(seconds);
   return `${hms} interval (${alignment?.text ?? ''})`;
 };
+
+export const getMetricType = (query?: TimeSeriesList) => {
+  const metricTypeKey = query?.filters?.findIndex((f) => f === 'metric.type')!;
+  const metricType = query?.filters?.[metricTypeKey + 2];
```

Comment:
I'm unsure on why we're adding 2 to the index here, maybe a comment for clarity?

Reply:
the filters are formed like `["metric.type", "=", "foo"]`, so if we have find the key `metric.type`, we need to add two to get the value (and skip the symbol). Let me add some test too for clarity.